web/src/lib/frontend-api.ts

```diff
@@ -63,10 +64,12 @@ export const restAPIRequest = async <T>(
  */
 export const graphQLRequest = async <T>(
   queryOptions: QueryOptions,
-  customErrorHandling?: boolean
+  customErrorHandling?: boolean,
+  headers?: Record<string, string>
```

Comment:
would using the Headers type work here?

Reply:
I'd leave as it is for now. We are passing it everywhere as object, Headers have different typing, using it make ts throw errors